modules/opengl/glsl/utils/shading.glsl

```diff
@@ -85,6 +85,10 @@ vec3 shadeSpecular(LightParameters light_, vec3 materialSpecularColor, vec3 posi
 
 vec3 shadeBlinnPhong(LightParameters light_, vec3 materialAmbientColor, vec3 materialDiffuseColor,
                      vec3 materialSpecularColor, vec3 position, vec3 normal, vec3 toCameraDir) {
+    // important for negative gradients/normals (inside/outside illumination)
+    if (dot(normal, toCameraDir) < 0.0) {
```

Comment:
Use faceforward
https://registry.khronos.org/OpenGL-Refpages/gl4/html/faceforward.xhtml

Reply:
faceforward for volume gradients?